app/api/metrics.py

```diff
@@ -51,3 +63,61 @@ def getScoreCards(
     teamId = decodeJWT(token)["teamId"]
 
     return microserviceScoreCardService.getByTeamId(teamId)
+"""
+
+# ADD NEW METRIC HERE
+@router.post("/", response_model=schemas.Metric)
+def createMetric(metric: schemas.MetricCreate, metricCrud: crud.CRUDMetric = Depends(dependencies.getMetricsCrud)) -> schemas.Metric:
+    metricObj = metric
+    # change ' ' with '-'
+    metricObj.code = format_code(metric.name)
+    # Stringfying the list of areas to be saved as string 
+    metricObj.area = json.dumps(metric.area)
+    # Handling the datatype of the field type to be integer or boolean 
+    if (metricObj.type != "integer" and metricObj.type != "boolean"):
+        raise ExceptionCustom(status_code=422, detail= "type must be integer or boolean")
+    metricCrud.create(metricObj)
+    raise ExceptionCustom(status_code=200, detail="Success in creating metric")
+
+# Get Any Metric Here By ID with parsing the list to be a real list not just stringified
+@router.get("/{metricID}", response_model=schemas.List[schemas.Metric])
+def getMetric(metricID: int, metricCrud: crud.CRUDMetric = Depends(dependencies.getMetricsCrud)) -> Any:
+    metric = metricCrud.get(metricID)
+    metric.area = json.loads(metric.area)
+    metricOBJ = jsonable_encoder(metric)
+    raise ExceptionCustom(status_code=200, detail=metricOBJ)
+
+# Delete any metric using its own ID
+@router.delete("/{metricID}")
+def deleteMetric(metricID: int, metricCrud: crud.CRUDMetric = Depends(dependencies.getMetricsCrud)) -> Any:
+    # this line is used to check if the metric is found to be deleted as
+    # if it is not found this will auto raise an error "Not Found"
+    metric = metricCrud.get(metricID)
+    metricCrud.delete(metricID)
+    raise ExceptionCustom(status_code = 200 , detail="deleted successfully")
+
+@router.put("/{metricID}")
+def editMetric(metricID: int, metricInput: schemas.MetricUpdate ,metricCrud: crud.CRUDMetric = Depends(dependencies.getMetricsCrud)) -> Any:
+    metric = metricCrud.get(metricID)
+    metricObj = metricInput
+    if (metricInput.name):
+        metricObj.name = metricInput.name
+        metricObj.code = format_code(metricInput.name)
```

Comment:
praise: good catch and nice work to put a constraint on the DB column!
make sure to handle the DB exception and rethrow an ExceptionCustom with a descriptive error message.